charts/community-operator/templates/database_roles.yaml

```diff
@@ -4,7 +4,10 @@ kind: ServiceAccount
 metadata:
   name: {{ .Values.database.name }}
   namespace: {{ if .Values.database.namespace }} {{ .Values.database.namespace }} {{ else }} {{ .Release.Namespace }} {{ end }}
-
+  {{- with .Values.database.serviceAccount.annotations }}
+  annotations:
```

Comment:
Why do we need it here? What's the use case?

Reply:
In often scenarios, specially with EKS, annotations are used to bind IAM roles to the serviceaccounts. One of the usecases can be for providing any additional permissions to the database serviceaccount. 

In my usecase, it was to allow IRSA for the database pods' serviceaccount, which has to retrieve the user's password from AWS Secrets Manager, and automatically create the secret using the [secret-store-csi-driver](https://secrets-store-csi-driver.sigs.k8s.io/)  with `extraObjects`